src/libcmd/installables.cc

```diff
@@ -514,6 +514,61 @@ ref<Installable> SourceExprCommand::parseInstallable(
     return installables.front();
 }
 
+static StorePath getBuiltPathFromRealisation(
+    Store & store,
+    const std::map<std::string, Hash> & outputHashes,
+    const DerivationOutputsAndOptPaths & drvOutputs,
+    const StorePath & drvPath,
+    const std::string & output)
+{
+    if (!outputHashes.count(output))
+        throw Error(
+            "the derivation '%s' doesn't have an output named '%s'",
+            store.printStorePath(drvPath), output);
+    if (experimentalFeatureSettings.isEnabled(Xp::CaDerivations)) {
+        auto outputId =
+            DrvOutput{outputHashes.at(output), output};
+        auto realisation =
+            store.queryRealisation(outputId);
+        if (!realisation)
+            throw Error(
+                "cannot operate on an output of unbuilt "
+                "content-addressed derivation '%s'",
+                outputId.to_string());
+        return realisation->outPath;
+    } else {
+        // If ca-derivations isn't enabled, assume that
+        // the output path is statically known.
+        assert(drvOutputs.count(output));
+        assert(drvOutputs.at(output).second);
+        return *drvOutputs.at(output).second;
+    }
+}
+
+static SingleBuiltPath getBuiltPath(ref<Store> evalStore, ref<Store> store, const SingleDerivedPath & b)
+{
+    return std::visit(
+        overloaded{
+            [&](const SingleDerivedPath::Opaque & bo) -> SingleBuiltPath {
+                return SingleBuiltPath::Opaque { bo.path };
+            },
+            [&](const SingleDerivedPath::Built & bfd) -> SingleBuiltPath {
+                auto drvPath = getBuiltPath(evalStore, store, *bfd.drvPath);
+                auto drv = evalStore->readDerivation(drvPath.outPath());
+                auto outputHashes = staticOutputHashes(*evalStore, drv); // FIXME: expensive
```

Comment:
Is this a dealbreaker or just something to do later?

Reply:
Oh I think this was coppied some fromewhere else hmm.